provision/provision-site.sh

```diff
@@ -30,12 +30,17 @@ function get_config_value() {
 
 function get_hosts() {
   local value=$(shyaml get-values-0 "sites.${SITE_ESCAPED}.hosts" 2> /dev/null < ${VVV_CONFIG} | tr '\0' ' ' | sed 's/ *$//')
-  echo "${value:-$@}"
+  echo "${value:-"${VVV_SITE_NAME}.test"}"
+}
+
+function get_hosts_list() {
+  local value=$(shyaml get-values "sites.${SITE_ESCAPED}.hosts" 2> /dev/null < ${VVV_CONFIG})
```

Comment:
Can we do this the same way as `if folders=$(shyaml keys -y -q "sites.${SITE_ESCAPED}.folders" < "${VVV_CONFIG}"); then` in the folders setup?

Reply:
also no `/dev/null`

Reply:
Replaced `/dev/null` with `-q` . I think the if structure here might overcomplicate the syntax, since basically the return statement returns the default if `value` is empty.

Reply:
it'd avoid certain issues with traps and if `shyaml` returned a bad error code. Also do we need the `-q`? If `shyaml` encounters a problem we'll be unable to debug it if it's silenced

Reply:
The problem could be simply that the path is not defined, which is why we default to the sitename. Afaik there's no default parameter for a list type of return of shyaml.

Can you elaborate on the traps issues? I'm not aware of what you mean.

Reply:
So originally I needed to know how to get a list of items but safely handle the situation when the path didn't exist, and the proposed method is what I got back from the shyaml author

Reply:
So what would be the errors that shyaml would throw that is not a path error?

Reply:
I'm not sure of the full list, but YAML syntax errors, and outright crashes in Shyaml itself could happen